### ISSUE TITLE
Can't assign "params" to "state"

### ISSUE THREAD

**Issue by PatrickJS** [BuilderIO/mitosis#388]
we need a way to assign `props` to `state`
https://github.com/BuilderIO/mitosis/blob/main/docs/gotchas.md#cant-assign-params-to-state

we can either inject the workaround into onMount or for some generators

**Comment by steve8708** [comment]
One thing I've been thinking we need that would help this is something like an `onInit` hook, like something that lets you run code immediately, before mount, e.g. like

```tsx
export default function MyComponent(props) {
  const state = useState({ foo: null })

   onInit(() => { state.foo = props.foo })
}
```

which then outputs to like `ngOnInit` for angular (or maybe constructor), `created` or `beforeCreate` for Vue, etc

I think `onMount` works in the meantime tho, some things occasionally may double render and it won't affect server side rendering, and we can add something like `onInit` (open to other names) for things that need to affect the server side / static render and ideally happen before render for other reasons

**Comment by PatrickJS** [comment]
yeah maybe we should change onInit to work sooner for this case

**Comment by PatrickJS** [comment]
actually if we just require this to happen in onInit then we can avoid all this. Ideally any prop on state can be refactored into onInit then when we generate the file the onInit code will be placed correctly

**Comment by steve8708** [comment]
Oh funny I didn't realize we already had it, but yeah we can do all this setup in that if in the right place

**Comment by PatrickJS** [comment]
Yeah I added it recently but it should definitely be used to set default values and nothing async
